pkgs/build-support/deno/buildDenoApplication.nix

```diff
@@ -0,0 +1,474 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchDenoDeps,
+  unzip,
+  xxd,
+  util-linux,
+  glibc,
+  stdenv,
+  fetchurl,
+  deno,
+  jq,
+}@topLevelArgs:
+{
+  # Name of the package
+  name ? "${args.pname}-${args.version}",
+  # Use this deno binary
+  deno ? topLevelArgs.deno,
+
+  # Name for the vendored dependencies tarball
+  denoDepsName ? "${name}-deno-deps",
+  # The vendored dependencies. The output structure of that derivation is still tbd
+  denoDeps ? (
+    fetchDenoDeps {
+      inherit
+        src
+        srcs
+        sourceRoot
+        prePatch
+        patches
+        postPatch
+        ;
```

Comment:
It might be possible to reduce this noise by using `builtins.intersectAttrs (builtins.functionArgs fetchDenoDeps) args` somewhere futher down.